src/overlays/actors/ovl_En_Grasshopper/z_en_grasshopper.c

```diff
@@ -15,41 +16,114 @@ void EnGrasshopper_Destroy(Actor* thisx, PlayState* play);
+void EnGrasshopper_RoamInCircles(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupBank(EnGrasshopper* this);
+void EnGrasshopper_Bank(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupBounced(EnGrasshopper* this);
+void EnGrasshopper_Bounced(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupApproachPlayer(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_ApproachPlayer(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupAttack(EnGrasshopper* this);
+void EnGrasshopper_Attack(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupWaitAfterAttack(EnGrasshopper* this);
+void EnGrasshopper_WaitAfterAttack(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_Damaged(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_Dead(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupFall(EnGrasshopper* this);
+void EnGrasshopper_Fall(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_InitializeEffect(EnGrasshopper* this, Vec3f* pos);
+void EnGrasshopper_UpdateEffects(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_DrawEffects(EnGrasshopper* this, PlayState* play);
+
+typedef enum {
+    /*  0 */ EN_GRASSHOPPER_ACTION_FLY,
+    /*  1 */ EN_GRASSHOPPER_ACTION_ROAM_IN_CIRCLES,
+    /*  2 */ EN_GRASSHOPPER_ACTION_BANK,
+    /*  3 */ EN_GRASSHOPPER_ACTION_BOUNCED,
+    /*  4 */ EN_GRASSHOPPER_ACTION_DECIDE_ACTION,
+    /*  5 */ EN_GRASSHOPPER_ACTION_APPROACH_PLAYER,
+    /*  6 */ EN_GRASSHOPPER_ACTION_ATTACK,
+    /*  7 */ EN_GRASSHOPPER_ACTION_WAIT_AFTER_ATTACK,
+    /*  8 */ EN_GRASSHOPPER_ACTION_DAMAGED,
+    /*  9 */ EN_GRASSHOPPER_ACTION_DEAD,
+    /* 10 */ EN_GRASSHOPPER_ACTION_FALL,
+} EnGrasshopperAction;
+
+typedef enum {
+    /* 0 */ EN_GRASSHOPPER_DECISION_ATTACK,
+    /* 1 */ EN_GRASSHOPPER_DECISION_FLY,
+    /* 2 */ EN_GRASSHOPPER_DECISION_ROAM_IN_CIRCLES, // Never used in the final game
+} EnGrasshopperNextAction;
+
+typedef enum {
+    /* 0 */ EN_GRASSHOPPER_BANK_STATE_BANKING,
+    /* 1 */ EN_GRASSHOPPER_BANK_STATE_DONE,
+} EnGrasshopperBankState;
+
+static s32 sOccupiedIndices[] = {
+    false, false, false, false, false,
+};
+
+static s8 sLimbIndexToShadowBodyPartsIndex[] = {
+    -1, -1, 0, 1, 2, 3, 4, 5, -1, -1, 6, 7, -1, 8, 9, -1, 10, 11, -1, 12, 13, -1, -1, -1,
+};
+
+static s8 sParentBodyParts[] = {
+    -1, -1, -1, 0, 1, 2, 3, -1, -1, -1, -1, 6, -1, -1, 8, -1, -1, 10, -1, -1, 12, -1, -1, -1,
+};
+
+static u8 sShadowSizes[] = {
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+};
+
+typedef enum {
+    /* 0x0 */ EN_GRASSHOPPER_DMGEFF_NO_DAMAGE,  // Deals no damage
+    /* 0x2 */ EN_GRASSHOPPER_DMGEFF_FIRE = 0x2, // Damages and sets the Dragonfly on fire
+    /* 0x3 */ EN_GRASSHOPPER_DMGEFF_FREEZE,     // Damages and freezes the Dragonfly in ice
+    /* 0x4 */ EN_GRASSHOPPER_DMGEFF_LIGHT_ORB,  // Damages and surrounds the Dragonfly with light orbs
```

Comment:
I think most of the enemies we've named it after the thing that causes the effect, i.e. fire/ice/light, rather than the actual thing it does. At least that's how I've been doing it.

Reply:
You gave me completely the opposite guidance on EnSnowman: https://github.com/zeldaret/mm/pull/818#discussion_r864344360
I've been following that guidance for every actor I've done from then to now, including this one. So which is actually preferred?

Reply:
Both have advantages and disadvantages, let's leave it and decide when we have all of them.

Reply:
Sounds good, and there will be inconsistencies anyways. Prior to Snowman, I named them after the *cause*. Then, from Snowman to present day, I named them after the *thing they do*. I never went back and changed my old actors to match how I've been doing things post-Snowman, so there will definitely be inconsistencies there. I guess the best thing to do now is "just give them a logical name and sort out the inconsistencies later".